### ISSUE TITLE
Can't start the container 

### ISSUE THREAD

**Issue by badyherz** [itzg/docker-minecraft-server#1895]
### Describe the problem

Having a container with the latest Minecraft version works fine.
But when I try to install a CurseForge modpack the container doesn't start, or at least it stops after some time.
I tried different things, but I'm very new to docker and server related stuff.
The only error code I found was with "docker inspect".

It seems like all the server related stuff and modpack stuff is fully installed without any errors.

I am open for any advice, thanks.

### Container definition

docker run -d -it -p 25565:25565 -v /home/badyherz/oneblock-docker:/data -e EULA=TRUE -e TYPE=AUTO_CURSEFORGE -e CF_SLUG=all-in-one-modded-one-block --name oneblockserver itzg/minecraft-server

### Container logs

"Health": {
                "Status": "unhealthy",
                "FailingStreak": 20,
                "Log": [
                    {
                        "Start": "2023-01-12T00:13:41.275078461Z",
                        "End": "2023-01-12T00:13:41.37966634Z",
                        "ExitCode": 1,
                        "Output": "failed to ping localhost:25565 : could not connect to Minecraft server: dial tcp 127.0.0.1:25565: connect: connection refused"
                    }

**Comment by itzg** [comment]
Since modpack is declared for 1.16.5 you'll need to run a Java 8 variant of the image such as `itzg/minecraft-server:java8-multiarch`. I'd tried it with that and seems to be running stable.

**Comment by badyherz** [comment]
Right, forgot that.
This fixes the crash of the container. It will keep running now.

But it still runs unhealthy with the same output:
```
 "ExitCode": 1,
                        "Output": "failed to ping localhost:25565 : could not connect to Minecraft server: dial tcp 127.0.0.1:25565: connect: connection refused"
```
Means I still can't connect to the server.
It is probably something in the modpack?
Because as I said, a container with the latest version worked fine.

I could try to create a blank forge 1.16.5 server to see any differences, if that would help.

**Comment by itzg** [comment]
It is most likely caused by the modpack; however, I tried the same and it seems to be up and running for me. Are there any manual modifications you made to server.properties such as `server-port` or `server-ip`?

**Comment by badyherz** [comment]
Not to my knowing. I just used the posted comment, changed nothing after that.

I tried the same mc version with the same forge version and it worked fine. Was up in a few minutes and I could join.
So it is the modpack. But what in there could it be? (Especially if it worked for you).

`docker run -d -it -p 25565:25565 -v /home/badyherz/test2:/data -e EULA=TRUE -e TYPE=FORGE -e VERSION=1.16.5 -e FORGE_VERSION=36.2.34 -e MEMORYSIZE='8G' --name mc2 itzg/minecraft-server:java8-multiarch`

**Comment by itzg** [comment]
Good find on the potentially related issue, but AUTO_CURSEFORGE should be immune to the problem there.

The only difference I can now think is that my test run might have grabbed a different file version of the modpack. I'll have to look later at which one it picked up. Can use `CF_FILE_ID` to pin the file version if that ends up being the thing.

**Comment by badyherz** [comment]
I tried different versions of the modpack (1.5.0;1.4.11;1.4.9) with `CF_FILE_ID` but same outcome. Some older versions crashed while startup.
Could there be a problem between AUTO_CURSEFORGE and duckdns?
Duckdns worked fine with a normal and empty forge server.

Is there any debugging I could do? Any logs, to find maybe more specific info?

**Comment by itzg** [comment]
> Could there be a problem between AUTO_CURSEFORGE and duckdns?
Duckdns worked fine with a normal and empty forge server.

Sorry, what is "duckdns" and how does it come into play?

**Comment by badyherz** [comment]
> Sorry, what is "duckdns" and how does it come into play?

"Duck DNS is a free service which will point a DNS (sub domains of duckdns.org) to an IP of your choice."
I use it so badyherzmc.duckdns.org can be used to join my local server.
So: badyherzmc.duckdns.org>MyIP>LocalServer

**Comment by itzg** [comment]
That all sounds quite custom. Let's see if anyone else comes across this with suggestions.

**Comment by badyherz** [comment]
Further info I guess. After further testing and trying to understand the logs I found out that this is always the last thing that happens and then nothing else. The container doesn't stop, but the logs stop, so it must be inactive.
```
[13Jan2023 13:39:52.263] [main/INFO] [KubeJS Server/]: Scanning recipes...
[13Jan2023 13:39:52.270] [main/INFO] [net.minecraft.resources.SimpleReloadableResourceManager/]: Reloading ResourceManager: KubeJS Virtual Data Pack [low priority], Default, astralsorcery-1.16-1.16.5-1.13.12.jar, supermartijn642configlib-1.0.9-mc1.16.jar, endportalrecipe_1.16.5-2.5.jar, mcw-windows-2.0.3-mc1.16.5.jar, InsaneLib-1.4.2-mc1.16.5.jar, CTM-MC1.16.1-1.1.2.6.jar, Controlling-7.0.0.28.jar, Placebo-1.16.5-4.7.0.jar, Bookshelf-Forge-1.16.5-10.4.32.jar, buildinggadgets-1.16.5-3.8.2.jar, ForgivingVoid_1.16.5-5.2.1.jar, ProgressiveBosses-3.4.3-mc1.16.5.jar, MekanismGenerators-1.16.5-10.1.2.457.jar, Hwyla-forge-1.10.11-B78_1.16.2.jar, carryon-1.16.5-1.15.5.22.jar, JustEnoughResources-1.16.5-0.12.1.128.jar, XNetGases-1.16.5-2.3.9.jar, refinedstorage-1.9.17.jar, konkrete_forge_1.3.3_MC_1.16.2-1.16.5.jar, RSInfinityBooster-1.16.5-1.1+13.jar, mcw-bridges-2.0.2-mc1.16.5.jar, FarmersDelight-1.16.5-0.6.0.jar, randomlib-1.16.x-1.0.0.jar, silent-gear-1.16.5-2.6.30.jar, supermartijn642corelib-1.0.18-forge-mc1.16.5.jar, simplylight-1.16.5-1.4.2-build.32.jar, Botania-1.16.5-420.2.jar, managear-1.16.3-2.2.3.jar, gardenofglass-1.7.jar, portality-1.16.5-3.2.5.jar, curios-forge-1.16.5-4.0.5.3.jar, eidolon-0.2.7.jar, Patchouli-1.16.4-53.2.jar, GameStages-Forge-1.16.5-7.3.13.jar, collective-1.16.5-4.25.jar, blockcarpentry-1.16-0.4.0.jar, elevatorid-1.16.5-1.7.13.jar, oneblock-1.16.x-1.7.1.jar, ExtraStorage-1.16.5-1.6.1.jar, architectury-1.31.60.jar, AI-Improvements-1.16.5-0.4.0.jar, weirdinggadget-1.16.(2-4)-2.2.7.jar, observerlib-1.16.5-1.5.3.jar, productivebees-1.16.5-0.6.9.25.jar, light-overlay-5.8.1.jar, cloth-config-4.14.54-forge.jar, trashcans-1.0.15-forge-mc1.16.jar, the_bumblezone_forge-3.2.1+1.16.5.jar, FastLeafDecay-v25.2.jar, geckolib-forge-1.16.5-3.0.68.jar, rhino-forge-1605.1.5-build.75.jar, Cucumber-1.16.5-4.1.12.jar, ftb-library-forge-1605.3.4-build.90.jar, ftb-teams-forge-1605.2.3-build.40.jar, jei-1.16.5-7.7.1.152.jar, item-filters-forge-1605.2.5-build.9.jar, controlled_random-1.1.0.jar, abnormals_core-1.16.5-3.3.1.jar, buzzier_bees-1.16.5-3.0.3.jar, simple-rpc-1.16.5-3.0.0.jar, Mekanism-1.16.5-10.1.2.457.jar, shutupexperimentalsettings-1.0.3.jar, configured-1.5.1-1.16.5.jar, LibX-1.16.3-1.0.76.jar, decorative_blocks-1.16.4-1.7.2.jar, BotanyPots-1.16.5-7.1.33.jar, BotanyTrees-1.16.5-3.0.7.jar, charginggadgets-1.3.0.jar, TimeIsUp-1.16-1.5.2.jar, filteredchests-1.0.2.jar, TinyMobFarm-1.16.3-1.1.1.jar, mcjtylib-1.16-5.1.6.jar, rftoolsbase-1.16-2.1.3.jar, xnet-1.16-3.0.16.jar, questsadditions-1.16-1.3.0.jar, ars_nouveau-1.16.5-1.25.0.jar, ftb-chunks-forge-1605.3.2-build.115.jar, kubejs-forge-1605.3.19-build.299.jar, forge-1.16.5-36.2.34-universal.jar, endergetic-1.16.5-3.0.2.jar, MysticalAgriculture-1.16.5-4.2.6.jar, ironchest-1.16.5-11.2.21.jar, ZeroCore2-1.16.5-2.1.16.jar, ExtremeReactors2-1.16.5-2.0.43.jar, TConstruct-1.16.5-3.3.2.324.jar, MouseTweaks-2.14-mc1.16.2.jar, titanium-1.16.5-3.2.8.4-10.jar, ftb-quests-forge-1605.3.6-build.98.jar, ImmersiveEngineering-1.16.5-5.0.7-143.jar, silent-lib-1.16.3-4.9.6.jar, SkyblockBuilder-1.16.4-1.6.14.jar, movingelevators-1.3.7-forge-mc1.16.jar, rftoolsbuilder-1.16-3.1.8.jar, jeiintegration_1.16.5-7.0.1.15.jar, flywheel-1.16-0.2.5.jar, create-mc1.16.5_v0.3.2g.jar, Mantle-1.16.5-1.6.157.jar, Xaeros_Minimap_22.6.0_Forge_1.16.5.jar, gravestone-1.16.5-1.0.6.jar, ftb-backups-2.1.2.2.jar, FastWorkbench-1.16.5-4.6.1.jar, polymorph-forge-1.16.5-0.40.jar, JustEnoughProfessions-1.16.5-1.2.2.jar, AutoRegLib-1.6-49.jar, Quark-r2.4-322.jar, StorageDrawers-1.16.3-8.5.2.jar, fancymenu_forge_2.7.0_MC_1.16.2-1.16.5.jar, FastFurnace-1.16.5-4.5.0.jar, appleskin-forge-mc1.16.x-2.4.0.jar, Aquaculture-1.16.5-2.1.22.jar, SilentGems-1.16.3-3.7.16.jar, chiselsandbits-1.0.63.jar, loadmyresources_1.0.1_MC_1.16.2-1.16.5.jar, KubeJS Resource Pack [data], KubeJS Virtual Data Pack [high priority]
[13Jan2023 13:40:02.734] [Worker-Main-11/DEBUG] [mixin/]: Mixing TagBuilderMixin from kubejs-common.mixins.json into net.minecraft.tags.ITag$Builder
[13Jan2023 13:40:31.437] [Worker-Main-10/DEBUG] [botania/]: Read botania:orechid.json data files in 5.622 ms
[13Jan2023 13:40:31.438] [Worker-Main-10/DEBUG] [botania/]: Read botania:orechid_ignem.json data files in 217.6 Î¼s
[13Jan2023 13:41:24.460] [Worker-Main-12/ERROR] [net.minecraft.client.resources.JsonReloadListener/]: Couldn't parse data file minecraft:recipes/building_blocks/purpur_stairs from minecraft:advancements/recipes/building_blocks/purpur_stairs.json
java.io.FileNotFoundException: minecraft:advancements/recipes/building_blocks/purpur_stairs.json
	at net.minecraft.resources.SimpleReloadableResourceManager.func_199002_a(SimpleReloadableResourceManager.java:63) ~[?:?]
	at net.minecraft.client.resources.JsonReloadListener.func_212854_a_(JsonReloadListener.java:44) ~[?:?]
	at net.minecraft.client.resources.JsonReloadListener.func_212854_a_(JsonReloadListener.java:22) ~[?:?]
	at net.minecraft.client.resources.ReloadListener.func_215270_b(SourceFile:11) ~[?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1604) [?:1.8.0_312]
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1596) [?:1.8.0_312]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) [?:1.8.0_312]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) [?:1.8.0_312]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) [?:1.8.0_312]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175) [?:1.8.0_312]

```
The last part with the error repeats a lot of times with different items, and then the log stops.

Container command:
`docker run -d -it -p 25565:25565 -v /home/badyherz/debug1:/data -e DEBUG=TRUE -e EULA=TRUE -e TYPE=AUTO_CURSEFORGE -e CF_SLUG=all-in-one-modded-one-block -e MEMORYSIZE='8G' --name debug1 itzg/minecraft-server:java8-multiarch`

**Comment by itzg** [comment]
I'm not sure what version mine picked up before, because now I hit the same errors you posted. 

It seems like something is wrong with tconstruct and some other mod incompatibility. You'll probably need to tweak the exclude/include options.

**Comment by itzg** [comment]
...or, I would suggest reporting an issue https://www.curseforge.com/minecraft/modpacks/all-in-one-modded-one-block/issues

**Comment by badyherz** [comment]
Well.. I don't think I need to report it.

> BTW, the variable is MEMORY not MEMORYSIZE
Turns out that was my mistake all the time. I got everything working now.
I did not really think about memory issues. Modpacks need more memory than standard.
`MEMORY='8G' ` seems to be working fine with this modpack.

I used `MEMEORYSIZE `because another container used it.

As info for others, this should work fine:
`docker run -d -it -p 25565:25565 -v /home/user/minecraft:/data -e EULA=TRUE -e TYPE=AUTO_CURSEFORGE -e CF_SLUG=all-in-one-modded-one-block -e MEMORY='8G' --name mc itzg/minecraft-server:java8-multiarch`

Thanks for the fast reply and help.
I'll close this issue with this comment.
